c/ckb_identity.h

```diff
@@ -352,6 +501,93 @@ int verify_sighash_all(uint8_t *pubkey_hash, uint8_t *sig, uint32_t sig_len,
   return 0;
 }
 
+static void bin_to_hex(const uint8_t *source, uint8_t *dest, size_t len) {
+  const static uint8_t HEX_TABLE[] = {'0', '1', '2', '3', '4', '5', '6', '7',
+                                      '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
+  for (int i = 0; i < len; i++) {
+    dest[i * 2] = HEX_TABLE[source[i] >> 4];
+    dest[i * 2 + 1] = HEX_TABLE[source[i] & 0x0F];
+  }
+}
+
+int convert_btc_message_variant(const uint8_t *msg, size_t msg_len,
+                                uint8_t *new_msg, size_t new_msg_len,
+                                const char *magic, const uint8_t magic_len) {
+  if (msg_len != new_msg_len || msg_len != SHA256_SIZE)
+    return ERROR_INVALID_ARG;
+
+  uint8_t temp[MESSAGE_HEX_LEN];
+  bin_to_hex(msg, temp, 32);
```

Comment:
Can we pass in binary directly? Will that result in problems?

Reply:
For some reason, some BTC wallets support strings to be passed in. 
Also in PW wallet, it uses same strategy.
For modern blockchain like eth, it can be directly passed in binary.

Reply:
<img width="1048" alt="image" src="https://github.com/nervosnetwork/ckb-production-scripts/assets/265443/545de35a-5803-4163-af3d-5465190fd608">

See UniSat wallet signing. It can't display binary values.